### PR TITLE
SCF/Assignment6

### PR DESCRIPTION
**Pull request by meta-nilesh-patel**

### CODE REVIEW

File: SCF/Session 6/Assignment6B/Poly.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+{
+	private final int[] polynomial;
+	
+	/**
+	 * 
+	 * Parameterized Constructor of class
+	 * @param int array of Polynomial
+	 * @throws IOException if the poly array is null
+	 */
+	Poly(int[] poly)throws IOException
```
Comment (rash-tiwari) [inline SCF/Session 6/Assignment6B/Poly.java]: Add space before ```throws```

File: SCF/Session 6/Assignment6B/Poly.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	 * Parameterized Constructor of class
+	 * @param int array of Polynomial
+	 * @throws IOException if the poly array is null
+	 */
+	Poly(int[] poly)throws IOException	
+	{	
+		//inputValidation
+		if(poly.length == 0)
+		{
+			throw new IOException("Null Polynomial Not Allowed");
```
Comment (rash-tiwari) [inline SCF/Session 6/Assignment6B/Poly.java]: Replace all ```IOException``` with either AssertionError or Exception

File: SCF/Session 6/Assignment6B/Poly.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import java.io.IOException;
+
+/**
+ * 
+ * @author Nilesh Patel
+ * Immutable class for Polynomial Operations
+ */	
+final class Poly
```
Comment (rash-tiwari) [inline SCF/Session 6/Assignment6B/Poly.java]: Add ```public``` aceess specifier before class

File: SCF/Session 6/Assignment6B/Poly.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	 * @param poly2 : Object of Poly Class of Second Value
+	 * @return Poly type of product of polynomials
+	 * throws IOException if array of polynomial is null
+	 */
+	public static Poly multiplyPoly(Poly poly1, Poly poly2)throws IOException 
+	{
+		int[] result = new int[poly1.polynomial.length + poly2.polynomial.length - 1];
+		for(int indexFirst = poly1.polynomial.length - 1; indexFirst >= 0; indexFirst--) 
+		{
+			for(int indexSecond = poly2.polynomial.length - 1;indexSecond >= 0; indexSecond--)
```
Comment (rash-tiwari) [inline SCF/Session 6/Assignment6B/Poly.java]: Add space after each operator

File: SCF/Session 6/Assignment6B/TestAddPolyMethod.java [hunk 5]
```diff
@@ -73,7 +73,7 @@ public void teat2()
 		{
 			Poly objectOfPolyClass1 = new Poly(new int[]{});
 			Poly objectOfPolyClass2 = new Poly(new int[]{4, 5, 6});
-			int[] actualOutput = Poly.addPoly(objectOfPolyClass1, objectOfPolyClass2).getPolynomial();
+			int[] actualOutput = Poly.addPoly(objectOfPolyClass1, objectOfPolyClass2).getPolynomialArray();
 		}
 		catch(IOException io)
```
Comment (rash-tiwari) [inline SCF/Session 6/Assignment6B/TestAddPolyMethod.java]: Add ```Exception``` in place of ```Exception```

**Review by rash-tiwari — CHANGES_REQUESTED** [review]
Code Review Done at 1st level

File: SCF/Session 6/Assignment6B/Poly.java (hunk 1 above)
Comment (meta-nilesh-patel) [reply SCF/Session 6/Assignment6B/Poly.java]: Done

File: SCF/Session 6/Assignment6B/Poly.java (hunk 2 above)
Comment (meta-nilesh-patel) [reply SCF/Session 6/Assignment6B/Poly.java]: Done

File: SCF/Session 6/Assignment6B/Poly.java (hunk 3 above)
Comment (meta-nilesh-patel) [reply SCF/Session 6/Assignment6B/Poly.java]: Done

File: SCF/Session 6/Assignment6B/Poly.java (hunk 4 above)
Comment (meta-nilesh-patel) [reply SCF/Session 6/Assignment6B/Poly.java]: Done

File: SCF/Session 6/Assignment6B/TestAddPolyMethod.java (hunk 5 above)
Comment (meta-nilesh-patel) [reply SCF/Session 6/Assignment6B/TestAddPolyMethod.java]: Done

**Review by meta-nilesh-patel — COMMENTED** [review]
Done

**Review by rash-tiwari — APPROVED** [review]

**Pull request merged**